snakemake_executor_plugin_slurm/job_status_query.py

```
@@ -155,3 +158,54 @@ def query_job_status_squeue(runid) -> list:
+        logger: Logger instance for debug/error output
+
+    Returns:
+        Tuple of (status_dict, query_duration) where status_dict is a dict
+        mapping job IDs to status strings, and query_duration is time in seconds.
+        Returns (None, None) on query failure.
+    """
+    status_of_jobs = {}
+
+    start_time = time.time()
+    try:
+        process = subprocess.Popen(
```

Ruff (0.15.1)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)